src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java

```diff
@@ -159,14 +162,18 @@ private static Path rebuildModuleHierarchy(Map<String, String> properties, Map<M
     for (String modulePathStr : current.getModules()) {
       File modulePath = new File(current.getBasedir(), modulePathStr);
       MavenProject module = findMavenProject(modulePath, propsByModule.keySet());
-      if (module != null) {
-        String moduleId = module.getGroupId() + ":" + module.getArtifactId();
-        Path topLevelModuleDir = rebuildModuleHierarchy(properties, propsByModule, module, prefix + moduleId + ".");
-        moduleIds.add(moduleId);
-        if (!topLevelModuleDir.startsWith(topLevelDir)) {
-          // Find common prefix
-          topLevelDir = findCommonParentDir(topLevelDir, topLevelModuleDir);
-        }
+      if (module == null) {
+        continue;
+      }
+      String moduleId = module.getGroupId() + ":" + module.getArtifactId();
+      if (specifiedProjectKey != null) {
+        moduleId = specifiedProjectKey + ":" + moduleId;
```

Comment:
The moduleId is only used to uniquely identify each module during the scan. I think you can keep the previous behavior, it will limit the chances to affect users.